Added async step execution for map operation (#410)
RedisGears/RedisGears: docs/runtime.md

```python
{{ include('runtime/log.py') }}
```

## gearsFuture
The `gearsFuture()` function is imported to the runtime's environment by default.

> This function returns a `gearsFuture` object which indicates that the record was taken to be processed on another thread/process. By returning this object from the step operation, RedisGears knows that the execution needs to hold until the background processing will be finished/failed. The `gearsFuture` expose two functions, `continueRun` and `continueFailed`, to continue/failed the processing (those 2 are thread-safe and can be called immediately when the processing has done). `continueRun` gets the record to continue the processing with, while `continueFailed` get a string representation of the failure description.

Reviewer: ```suggestion
This function returns a `gearsFuture` object, which allows another thread/process to process the record. Returning this object from a step's operation tells RedisGears to suspend execution until background processing had finished/failed.

The `gearsFuture` object provides two control methods: `continueRun()` and `continueFailed()`. Both methods are thread-safe and can be called at any time to signal that the background processing has finished. `continueRun` signals success and its argument is a record for the main process. `continueFailed` reports a failure to the main process and its argument is a string describing the failure.
```

> The following steps support `gearsFuture()`:

Reviewer: ```suggestion
Calling `gearsFuture()` is supported only from the context of the following operations:
```

> * map
> * flatmap
> * filter
> * foreach
> * aggregate
> * aggregateby

Reviewer: ```suggestion
* `map`
* `flatmap`
* `filter`
* `foreach`
* `aggregate`
* `aggregateby`
```

> Any attempt to create a `gearsFuture` outside of those steps will results in exception.

Reviewer: ```suggestion
An attempt to create a `gearsFuture` object outside of the supported contexts will result in an exception.
```

**Examples**

```python
{{ include('runtime/gearsFuture.py') }}
```

### gearsFuture with Python Async Await

> `gearsFuture` is also supported seamlessly with python async await, so it possible to do the following:

Reviewer: ```suggestion
`gearsFuture` is also integrated seamlessly with Python's async/await syntax, so it possible to do the following:
```
